File: sample/src/main/java/net/protoqueue/sample/SampleProtoCore.java
package net.protoqueue.sample;

import net.protoqueue.ProtoDisposable;

/**
 * Created by zhongyongsheng on 2018/4/20.
 */
public class SampleProtoCore {
    private static final String TAG = "SampleProtoCore";
    private long topSid = 100000;
    private long subSid = 100000;
    int appId = 10086;
    ProtoDisposable mDisposable;

    private SampleProtoQueue sampleProtoQueue = SampleProtoQueue.getInstance();

    public void sendGameListReq() {
        mDisposable = sampleProtoQueue.sendSampleProto();
    }

    public void mockOnReceive() {
        SampleProto sampleProto = new SampleProto(new byte[]
                {11, sampleProtoQueue.getSeqContext().byteValue(), 100});
        sampleProtoQueue.onReceiveData(appId, sampleProto.toByteArray());
    }

    public void dispose() {
        if (mDisposable != null)
            mDisposable.dispose();
    }
}

File: sample/src/main/java/net/protoqueue/sample/simple/SampleProtoCore.kt
package net.protoqueue.sample.simple

import net.protoqueue.ProtoDisposable

/**
 * Created by zhongyongsheng on 2018/4/20.
 */
class SampleProtoCore {
    private val topSid: Long = 100000
    private val subSid: Long = 100000
    internal var appId = 10086
    internal var mDisposable: ProtoDisposable? = null


    private val sampleProtoQueue = SampleProtoQueue.instance

    fun sendGameListReq() {
        mDisposable = sampleProtoQueue.sendSampleProto()
    }

    fun sendGameListReqByQueueParameter() {
        mDisposable = sampleProtoQueue.sendSampleProtoByQueueParameter()
    }

    fun mockOnReceive() {
        val sampleProto = SampleProto(
            byteArrayOf(11, sampleProtoQueue.getSeqContext().toByte(), 100))
        sampleProtoQueue.onReceiveData(appId, sampleProto.toByteArray())
    }

    fun dispose() {
        mDisposable?.dispose()
        sampleProtoQueue.cancel()
    }

    fun sendGameListReqInCoroutine() {
        sampleProtoQueue.sendSampleProtoInCoroutine()
    }

    companion object {
        private val TAG = "SampleProtoCore"
    }

}
